xray/core/dataarray.py

```diff
@@ -1091,6 +1091,16 @@ def plot(self):
         '''
         return _PlotMethods(self)
 
+    def _title_for_slice(self):
```

Comment:
To me, this logic seems more appropriate for a helper function in the plotting module (closer to where it is called).